### PR TITLE
Suggested proto changes for PBBV0-306

### PR DESCRIPTION
**Pull request by endrec**
https://kodypay.atlassian.net/browse/PBBV0-306

### CODE REVIEW

File: src/main/proto/com/kodypay/grpc/pay/v1/pay.proto [hunk 1]
```diff
@@ -92,3 +93,120 @@ message RefundResponse {
     REFUNDED = 3;
   }
 }
+
+// requires X-API-Key header with 'API Key' value
+message AuthorizationRequest {
+  string store_id = 1;                    // UUID of store
+  string amount = 2;                      // Amount in BigDecimal/2.dp (0.00)
+  string terminal_id = 3;                 // Send the payment to this terminal serial number
+  optional string transToken = 4;         // Optional, max 60 chars
```
Comment (rfontes17) [inline src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:102]: name should use underscore `trans_token`

Comment (rfontes17) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:102]: same for the rest

File: src/main/proto/com/kodypay/grpc/pay/v1/pay.proto [hunk 2]
```diff
@@ -92,3 +93,120 @@ message RefundResponse {
     REFUNDED = 3;
   }
 }
+
+// requires X-API-Key header with 'API Key' value
+message AuthorizationRequest {
```
Comment (sishbi) [inline src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:98]: @endrec Can I understand the reason why this is needed?
What would happen in the backend for this?

Comment (endrec) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:98]: In short, it's the first half of a standard to step payment (https://kodypay.atlassian.net/browse/PBBV0-189), so the backend should ask the terminal to authorise a transaction up to `amount`, and return the relevant details so at a later point we can capture from this authorization, or release it.
Bear in mind that Symphony uses this same transaction type to check the validity of a card, where `amount` is `0`.

Use case: you walk into the bar, and ask the bartender to open you tab with a £100.

Comment (sishbi) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:98]: So this is a `PreAuth` request?
I get that now.
A follow up question: are all of these parameters required?  They seem specific for the needs of Oracle instead parameters that fit into our data model

Comment (endrec) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:98]: > are all of these parameters required? They seem specific for the needs of Oracle instead parameters that fit into our data model

Some of them is optional (as marked), and I tried to strip out what I thought is really oracle specific, the ones here I believe could be useful for any integrator.

File: src/main/proto/com/kodypay/grpc/pay/v1/pay.proto [hunk 3]
```diff
@@ -92,3 +93,120 @@ message RefundResponse {
...
+  string store_id = 1;                    // UUID of store
+  string amount = 2;                      // Amount in BigDecimal/2.dp (0.00)
+  string terminal_id = 3;                 // Send the payment to this terminal serial number
+  optional string trans_token = 4;        // Optional, max 60 chars
```
Comment (sishbi) [inline src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:102]: What would be in the `trans_token` ?

Comment (sishbi) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:102]: These card-type fields are all in the request? So it implies that they have been captured in the POS ?

Comment (sishbi) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:102]: But we capture these in the terminal _normally_

Comment (sishbi) [reply src/main/proto/com/kodypay/grpc/pay/v1/pay.proto:102]: Or does this cater for manually entered card details? Again, something the terminal does...

**Comment by endrec** [comment]
[OPI - Auth.pdf](https://github.com/user-attachments/files/17775201/OPI.-.Auth.pdf)

**Pull request closed without merging**